src/random/cpu/sample_utils.h

```diff
@@ -32,6 +32,10 @@ class BaseSampler {
   }
 };
 
+// (BarclayII 2022.9.20) Changing the internal data type of probabilities to double since
+// we are using non-uniform sampling to sample on boolean masks, where False represents
```

Comment:
what's `DType` if not changed to `double`? could you explain more why change to double in order to handle boolean mask?

Reply:
if not fixed as `double`, `DType` could be `uint8` and this will cause type mismatch?

Reply:
Yes.  Let me elaborate in the comments.